Source/Dafny/Cloner.cs

```diff
@@ -601,9 +601,6 @@ class Cloner
       } else if (stmt is ForallStmt) {
         var s = (ForallStmt)stmt;
         r = new ForallStmt(Tok(s.Tok), Tok(s.EndTok), s.BoundVars.ConvertAll(CloneBoundVar), null, CloneExpr(s.Range), s.Ens.ConvertAll(CloneMayBeFreeExpr), CloneStmt(s.Body));
-        if (s.ForallExpressions != null) {
```

Comment:
When do expressions need to be cloned?

Reply:
Whenever there's a need to create a new syntactic copy of a part of an AST. Examples are:
* when a module is refined, its contents is cloned and included into the new module
* when the program is compiled, all of its modules get cloned before they are compiled (this used to be needed, but I'm not actually sure there's any reason to still do this)
* when the various export sets of a module are checked for consistency, the exported components are cloned
* when an `inductive predicate`, `copredicate`, `inductive lemma`, or `colemma` is syntactically desugared, parts of it are cloned